testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,323 @@
+                                        {"stringA0", 0, 100}
+                                }));
+
+        alterTableSchemaTo(REMOVED_COLUMN_SCHEMA);
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(new String[][] {
+                                {"int_col", "integer", "", ""}
+                        }));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        prepareQueryResultRows(
+                                new Object[][] {
+                                        {0}
+                                }));
+
+        alterTableSchemaTo(RENAMED_COLUMN_SCHEMA);
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(new String[][] {
+                                {"string_col", "varchar", "", ""},
+                                {"int_col_renamed", "integer", "", ""},
+                        }));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        prepareQueryResultRows(
+                                new Object[][] {
+                                        {"stringA0", null}
+                                }));
+    }
+
+    @Test(groups = AVRO)
+    public void testSchemaWhenUrlIsUnset()
+    {
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(new String[][] {
+                                {"string_col", "varchar", "", ""},
+                                {"int_col", "integer", "", ""}
+                        }));
+        assertThat(query(selectStarStatement))
+                .containsExactlyInOrder(
+                        prepareQueryResultRows(
+                                new Object[][] {
+                                        {"stringA0", 0}
+                                }));
+
+        onHive().executeQuery(format("ALTER TABLE %s UNSET TBLPROPERTIES('avro.schema.url')", tableName));
+        assertThat(query(columnsInTableStatement))
+                .containsExactlyInOrder(
+                        prepareShowColumnsResultRows(new String[][] {
+                                {"dummy_col", "varchar", "", ""}
+                        }));
+    }
+
+    protected void alterTableSchemaTo(String schema)
+    {
+        onHive().executeQuery(format("ALTER TABLE %s SET TBLPROPERTIES('avro.schema.url'='%s')", tableName, schema));
+    }
+
+    private void insertData(String tableName, int rowNo, String... sqlValues)
+    {
+        String columnValues = Joiner.on(", ").join(sqlValues) +
+                (partitions.isEmpty() ? "" : ", '" + getPartitionsAsString("', '", p -> p.getValueForRow(rowNo)) + "'");
+        query(format("INSERT INTO %s VALUES (%s)", tableName, columnValues));
+    }
+
```

Comment:
Can we have something like 
```
private Row createRow(int rowNumber, Object.. data)
{
    ImmutableList.Builder<Object> builder = ImmutableList..builder();
    builder.addAdd(data);
    partitions.forEach(partition -> builder.add(partition.getValueForRow(rowNo));
    return new Row(builder.build());
}
```

And we could have tests like 
```
        assertThat(query(columnsInTableStatement))
                .containsExactlyInOrder(
                                createRow(1, "string_col", "varchar", "", ""),
                                createRow(2, "int_col", "integer", "", ""),
                                createRow(3, "int_col_added", "integer", "", "")));

Reply:
We need to pass as `Row[]` instead we could pass it as `Row` and handle accordingly

Reply:
Done